llvm/test/CodeGen/AMDGPU/codegen-prepare-addrspacecast-non-null.ll

```diff
@@ -0,0 +1,279 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: opt -mtriple=amdgcn-- -amdgpu-codegenprepare -S < %s | FileCheck -check-prefix=OPT %s
+; RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx900 < %s | FileCheck %s --check-prefixes=ASM,DAGISEL-ASM
+; RUN: llc -mtriple=amdgcn-amd-amdhsa -global-isel -mcpu=gfx900 < %s | FileCheck %s --check-prefixes=ASM,GISEL-ASM
+
```

Comment:
Should also have a standalone codegen test for the intrinsic. Code suggests it may support vectors, so should include some vector tests

Reply:
I tried writing one but it's rejected, the intrinsic signature doesn't allow it:
```
Intrinsic has incorrect return type!
ptr @llvm.amdgcn.addrspacecast.nonnull.v4p0.v4p3
```